Automattic/wp-calypso
# Jetpack Pricing Cart Integration - Handle CTA Label and Cart change

#### Proposed Changes

- Changes to show the button label as **View Cart** with icon when an item is added to cart
- Clicking of **View Cart** should navigate to checkout page
- Add checks for CRM products and Bundles - which should remain as it was before
- Wrap the Pricing page component with `CartSubscriber` which listens to cart and forces re-render when cart changes

#### Testing Instructions

* click on Jetpack Cloud live link below
    * Goto `/pricing`
* or boot up this PR 
    * Run `git fetch && git checkout fix/fix/view-cart-cta`
    * Run `yarn start-jetpack-cloud`
    * Goto http://jetpack.cloud.localhost:3000/pricing
* Switch to Bundle section and confirm the button Label on the card is the same as before and confirm that clicking on it takes you to checkout page
* Switch to the Product section and confirm the following
    * All the product cards (except the one already purchased and CRM) CTA label should be  **Add to Cart** instead of **Get**
    * Clicking on the CTA adds the item to the cart in header and button label for the item changes to **View Cart**
    * Make sure the **View Cart** CTA matches the design
    * Click on **View Cart** should redirect to checkout page
    * Go back to pricing page and remove an item from cart and confirm that the CTA for that item changes back to **Add to Cart**
    * Confirm that for CRM CTA says **Get** and clicking on it takes to CRM page
  
 
#### Video

[Desktop-1670928897817.webm](https://user-images.githubusercontent.com/2027003/207299972-1fbfd911-dc50-445d-8c68-15b6e404bef0.webm)

#### Products page
| Before | After |
|--------|------|
| <img width="500" alt="Screen Shot 2022-12-06 at 3 26 40 PM" src="https://user-images.githubusercontent.com/2027003/207302375-72cbd55d-1296-4389-86cc-f02c5faa3f9a.png"> | <img width="500" alt="Screen Shot 2022-12-06 at 4 33 54 PM" src="https://user-images.githubusercontent.com/2027003/207302408-d8ad001c-af61-4a59-85d4-bee94333ce93.png"> |

#### Bundles page
| Before | After |
|--------|------|
| <img width="500" alt="Screen Shot 2022-12-06 at 3 26 40 PM" src="https://user-images.githubusercontent.com/2027003/207303347-a2756e3d-aa33-4072-9c3f-d453c9486c67.png"> | <img width="500" alt="Screen Shot 2022-12-06 at 4 33 54 PM" src="https://user-images.githubusercontent.com/2027003/207303363-fa276f92-95bc-4acc-ba6e-bff84ee3fda2.png"> |


#### Pre-merge Checklist






Related to #
1203495437860512-as-1203534015891085/f
1203495437860512-as-1203534015891165/f

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203534015891085
  - https://app.asana.com/0/0/1203534015891165
